lib/gollum/public/gollum/javascript/gollum.js.erb

```diff
@@ -658,6 +658,54 @@ $(document).ready(function() {
         }
       });
     }
+
+    if (navigator.clipboard) {
+      // Create copy to clipboard buttons for code-blocks
+      const codeBlocks = document.querySelectorAll("pre > code");
+      const iconCopyDefault = `<%=rocticon('copy', {:width => 18, :height => 18})%>`;
+      const iconCopyActive = `<%=rocticon('check', {:width => 18, :height => 18})%>`;
+
+      const copyToClipboard = async (event) => {
+        const { currentTarget: button } = event;
+
+        const codeBlock = document.querySelector(`#${button.dataset.codeblock}`);
+
+        if (!codeBlock) {
+          throw('Error while coping to clipboard: codeblock not found.');
+          return;
+        }
+
+        await navigator.clipboard.writeText(codeBlock.textContent);
+
+        button.innerHTML = iconCopyActive;
+        button.classList.add("tooltipped", "tooltipped-no-delay", "tooltipped-w");
+        button.setAttribute('aria-label', 'Copied!');
+
+        // wait 2 secconds
+        await new Promise(resolve => setTimeout(resolve, 2000));
+
+        button.innerHTML = iconCopyDefault;
+        button.classList.remove("tooltipped", "tooltipped-no-delay", "tooltipped-w");
+        button.setAttribute('aria-label', 'copy to clipboard');
+      };
+
+      codeBlocks.forEach((item, idx) => {
+        // assign a random & unique ID so the element can be selected later.
+        const id = `code-clipboard-${idx + 1}`;
+
+        const button = document.createElement("button");
+
+        button.classList.add("btn", "btn-octicon", "border", "position-absolute", "top-2", "right-2");
+        button.setAttribute('aria-label', 'copy to clipboard');
+        button.dataset.codeblock = id;
+        button.innerHTML = iconCopyDefault;
+        item.id = id;
+
+        button.onclick = copyToClipboard;
+
+        item.parentNode.appendChild(button);
```

Comment:
Not sure if this is the best way to do this, but adding the following should make sure that the buttons are only visible when the mouse is on the code block.
```javascript
        item.parentNode.addEventListener('mouseenter', function(element) {
          element.target.querySelector('.btn').classList.remove("d-none");
        });
        item.parentNode.addEventListener('mouseleave', function(element) {
          element.target.querySelector('.btn').classList.add("d-none");
        });
```
NB. The initial classlist of the button should then also include `"d-none"` to make the buttons invisible at the beginning.

Reply:
I used the hidden property as explained in the [prime.css documentation](https://primer.style/css/utilities/layout#the-html-hidden-attribute).

I also added a little fade-in animation in the CSS, prime.css have a class to add animations but have 1s duration and I thinks that's a little too much.

Reply:
> I used the hidden property as explained in the [prime.css documentation](https://primer.style/css/utilities/layout#the-html-hidden-attribute).

Much better. 👍